src/gtc/passes/oir_optimizations/horizontal_execution_merging.py

```diff
@@ -47,13 +47,26 @@ def visit_VerticalLoopSection(
         new_symbol_name: Callable[[str], str],
         **kwargs: Any,
     ) -> oir.VerticalLoopSection:
-        horizontal_executions = [node.horizontal_executions[0]]
-        new_block_extents = [block_extents[id(horizontal_executions[-1])]]
+        @dataclass
+        class UncheckedHorizontalExecution:
+            # local replacement without type checking for type-checked oir node
+            # required to reach reasonable run times for large node counts
+            body: List[oir.Stmt]
+            declarations: List[oir.LocalScalar]
+
+            def to_oir(self):
+                return oir.HorizontalExecution(body=self.body, declarations=self.declarations)
+
+        def to_unchecked(hexec):
+            return UncheckedHorizontalExecution(body=hexec.body, declarations=hexec.declarations)
```

Comment:
Nice work! This could be a classmethod of `UncheckedHorizontalExecution`. If leaving as is, then perhaps make another custom constructor to use for `horizontal_executions[-1] = UncheckedHorizontalExecution(...)` on line 103.

Reply:
Chose `from_oir` to match the `to_oir`. Also added an assertion to check for changes in `oir.HorizontalExecution` fields.